### PR TITLE
feat(dispatch): execution metrics — attempt-success % + per-DA pace

### PR DESCRIPTION
**Pull request by BanerjeeAgniva**
## What
Roadmap **step 2c** control-tower core — the delivery **attempt-success gauge** and **per-DA pace**, the two things Amazon's Route Execution console leads with.

`GET /dispatch/execution?date=` (STATION_MANAGER, city-scoped; ADMIN all cities; date defaults to today):

```json
{ "date": "2026-08-24",
  "attempt_success_pct": 0.9, "deliveries_completed": 18, "deliveries_failed": 2,
  "das": [
    { "da_id": "…", "stops_done": 5, "stops_last_hour": 2, "stops_pending": 6, "avg_per_hour": 1.0 },
    { "da_id": "…", "stops_done": 8, "stops_last_hour": 1, "stops_pending": 1, "avg_per_hour": 2.0 } ] }
```

## Design
- **Attempt-success** = `COMPLETED / (COMPLETED + FAILED)` DELIVERY tasks (null if no attempts).
- **Per-DA pace**: `stops_done` today, `stops_last_hour` (current pace), `stops_pending` (open), and `avg_per_hour` since first assignment — guarded so a fresh DA's single stop doesn't read as 60/hr. Busiest (most pending) first.
- **Pure DB aggregation** over `dispatch_queue` (`completed_at` is set on completion), so it's independent of the live in-memory tile authority — a new `DispatchMetricsService`, not coupled to `StationDispatchService`.
- Two native Postgres queries (FILTER + interval) via interface projections with **camelCase aliases** (matches the auth DA-summary pattern, so Postgres's lowercase fold binds to the getters).

## Verified
- dispatch metrics test 4 green (attempt-success, null-when-no-attempts, avg/hr + pending-sort, fresh-DA guard); controller test updated (7 green). App assembles.
- **Both native queries run against a real Postgres** (FILTER, interval, GROUP BY all valid).

## Follow-ups (rest of step 2c)
Projected-RTS (needs an ETA model per DA queue), DA scorecards over a window, CSV export. Web control-tower cards to render this are a separate frontend slice.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added dispatch execution statistics for a selected date, defaulting to today.
  * Displayed delivery attempt success rates, completed and failed deliveries, and per-agent pacing metrics.
  * Added access-controlled city or administrator scope to execution statistics.
  * Included pending stops, recent completions, and average hourly throughput for each delivery agent.

* **Tests**
  * Added coverage for metric calculations, sorting, edge cases, and database-backed execution reporting.

### CODE REVIEW

File: dispatch/src/main/java/com/oneday/dispatch/repository/DispatchQueueRepository.java [hunk 1]
```diff
@@ -65,4 +65,36 @@ and d.status in (com.oneday.dispatch.domain.TaskStatus.QUEUED,
...
+     * DA's first assignment time (for average-per-hour). Native (Postgres FILTER + interval).
+     * {@code city} null → all cities.
+     */
+    @Query(value = """
+            SELECT da_id AS daId,
+                   COUNT(*) FILTER (WHERE status = 'COMPLETED') AS done,
+                   COUNT(*) FILTER (WHERE status = 'COMPLETED'
+                                      AND completed_at >= now() - interval '1 hour') AS lastHour,
+                   COUNT(*) FILTER (WHERE status IN ('QUEUED', 'IN_PROGRESS')) AS pending,
+                   MIN(assigned_at) AS firstAssigned
```
[the remark this answers is not in the document]
Comment (BanerjeeAgniva) [reply dispatch/src/main/java/com/oneday/dispatch/repository/DispatchQueueRepository.java:94]: Verified empirically — declining this one. I added a real-Postgres `@DataJpaTest` (`DispatchQueueRepositoryTest.executionProjectionsBindAgainstRealPostgres`, commit 7ec86a0) that inserts rows and asserts every `DaPaceRow` getter (`getDaId`/`getLastHour`/`getFirstAssigned`) and `DeliveryOutcome` getter returns the right value. **It passes** — the unquoted camelCase aliases bind fine on this stack (Spring Data JPA + Hibernate 6 + Postgres does case-insensitive alias matching). This is also the exact pattern the shipped `auth.DaProfileRepository.findDaSummaries` / `findContactsByDaIds` use (`u.city_id AS cityId`, `u.id AS daId`), so quoting here would diverge from the established, working convention. Coverage added as requested.
